waku/v2/protocol/metadata/waku_metadata.go

```diff
@@ -225,14 +240,13 @@ func (wakuM *WakuMetadata) Connected(n network.Network, cc network.Conn) {
 		shard, err := wakuM.Request(wakuM.ctx, peerID)
 		if err == nil {
 			if shard == nil {
-				err = errors.New("no shard reported")
+				shouldDisconnect = false // No shard reported
```

Comment:
Here the logic is pretty confusing, is it possible to make it like this?
```
shard, err := wakuM.Request(wakuM.ctx, peerID)

if err != nil {
    logError()
    // disconnect from peers if error happens in request and they support the protocol
    if protocolSupported(err) {
        removeAndClosePeer()
    }
}

if shard != nil && shard.ClusterID != wakuM.clusterID {
    // disconnect if nodes in different cluster
     removeAndClosePeer()
}

return
```

In `Request` method, no longer need this
```
if isProtocolNotSupported(err) {
    return nil, nil
}
```
and can just return the error.

Reply:
I used a variant of your suggestion, as verifying for the protocol is no longer needed, since we are mounting waku_metadata only if the clusterID != 0, and then we expect that peers support the protocol, otherwise we can drop them.